File: GarbageV3-2025/app/src/main/java/dk/itu/GarbageV3/SortingActivity.java
package dk.itu.GarbageV3;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.io.IOException;

public class SortingActivity extends AppCompatActivity {

    //Shopping V1

    // Model: Database of items
    private static ItemsDB itemsDB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.find_item);

        ItemsDB.setContext(SortingActivity.this);

        itemsDB= ItemsDB.get();

        TextView items= findViewById(R.id.items);
        EditText sortInput= findViewById(R.id.find_item_to_sort);
        items.setText("Where to place item:");


        Button listItems= findViewById(R.id.list_button);
        listItems.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String itemWhat = sortInput.getText().toString();

                String itemWhere = itemsDB.findCategory(itemWhat);

                sortInput.setText(itemWhat + " should be placed in: " + itemWhere);

            }
        });
        sortInput.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                sortInput.setText("");
            }
        });

        Button addItem= findViewById(R.id.add_new_item_button);
        addItem.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(SortingActivity.this, AddItemActivity.class);
                startActivity(intent);
            }
        });

    }
}

File: GarbageV5-2025-Kotlin/app/src/main/java/dk/itu/garbagev5_2025_kotlin/SortingActivity.kt
package dk.itu.garbagev5_2025_kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.ComponentActivity
import androidx.activity.enableEdgeToEdge

class SortingActivity : ComponentActivity() {

    private lateinit var itemsDB: ItemsDB

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        setContentView(R.layout.find_item_activity)
        itemsDB = ItemsDB.getInstance(this)

        val titleWhereToSortItems: TextView = findViewById(R.id.title_where_to_sort_items)
        val findWhereToSortItemsButton: Button = findViewById(R.id.find_item_button)
        val sortInput: EditText = findViewById<EditText?>(R.id.find_item_to_sort)
        val addItemButton: Button = findViewById(R.id.add_new_item_button)


        findWhereToSortItemsButton.setOnClickListener{
            val itemWhat = sortInput.text.toString()
            val itemWhere = itemsDB.getWhere(itemWhat)
            sortInput.setText("Item ${itemWhat} should be placed in: ${itemWhere} container")
        }

        sortInput.setOnClickListener{
            sortInput.setText("")
        }

        addItemButton.setOnClickListener{
            val intent = Intent(this, AddItemActivity::class.java)
            startActivity(intent)
        }

    }
}